backtracking/subset_sum.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file
+ * @brief Program to Count number of subsets
+ *
+ * @details Subset problem (https://en.wikipedia.org/wiki/Subset_sum_problem)
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief subset sum algorithm
+ */
+namespace backtracking {
+/**
+ * @namespace Subsets
+ * @brief Functions for counting subsets in a given array with a given sum
```

Comment:
Would be useful to describe the algorithm and its complexity.

Reply:
Done